src/participant/prompts/intent.ts

```diff
@@ -0,0 +1,78 @@
+import * as vscode from 'vscode';
+
+import { getHistoryMessages } from './history';
+
+export type PromptIntent = 'Query' | 'Schema' | 'Docs' | 'Default';
+
+export class IntentPrompt {
+  static getAssistantPrompt(): vscode.LanguageModelChatMessage {
+    const prompt = `You are a MongoDB expert.
+Your task is to help guide a conversation with a user to the correct handler.
+You will be provided a conversation and your task is to determine the intent of the user.
+The intent handlers are:
+- Query
+- Schema
+- Docs
+- Default
+Rules:
+1. Respond only with the intent handler.
+2. Use the "Query" intent handler when the user is asking for code that relates to a specific collection.
```

Comment:
What is the user has not specified a namespace? We don't have information about a specific collection, but it still can be a query generation request.

Reply:
That's a good question, and raises a tradeoff we have to take. _Some_ users may ask a question that would result in code for a database, like how can I create a user with specific database permissions or what's the size of X database. In that case the /query handler would be forcing them to choose a collection which doesn't really make sense. I was aiming to make this prompt fairly intentionally selective so we don't restrict the user capabilities. The generic prompt has a lot of the capabilities of the /query resolver. 
Let me know if you think we could word this differently or guide the behavior to the query resolver differently!